### PR TITLE
fixes module output render

### PR DESCRIPTION
**Pull request by saurabhkg891**
fixes module output render of object values

[https://app.asana.com/0/1202464617347318/1203793856578606/f](https://app.asana.com/0/1202464617347318/1203793856578606/f)

### Output

image

### CODE REVIEW

File: radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx [hunk 1]
```diff
@@ -41,6 +41,26 @@ const renderTFValue = (val: string): JSX.Element => {
 }
 
 const renderTFValues = (val: any): JSX.Element => {
```
Comment (acolver) [inline radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx]: can we use `val: unknown` instead?

File: radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx [hunk 2]
```diff
@@ -41,6 +41,26 @@ const renderTFValue = (val: string): JSX.Element => {
...
+  if (typeof val === "object" && !Array.isArray(val)) {
+    return (
+      <>
+        {Object.keys(val).map((objKey: any) => (
```
Comment (acolver) [inline radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx]: `objKey: unknown`

File: radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx [hunk 3]
```diff
@@ -41,6 +41,26 @@ const renderTFValue = (val: string): JSX.Element => {
...
+          <div key={objKey}>
+            {typeof val[objKey] === "string" ? (
+              <div className="grid grid-cols-3 sm:grid-cols-2 gap-x-2 sm:gap-x-4 mt-1 break-words">
```
Comment (acolver) [inline radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx]: Shouldn't this be something like: `grid-cols-1 sm:grid-cols-2 md:grid-cols-3`

**Review by acolver — CHANGES_REQUESTED** [review]
We also need to check this function:
```ts
const renderTFValue = (val: string): JSX.Element
```

It may need to be changed to
```ts
const renderTFValue = (val: unknown): JSX.Element
```
And then do `if (typeof val === "string" && isURL(val)) ...`

because this is where the initial error was.

File: radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx [hunk 4]
```diff
@@ -33,19 +34,43 @@ const renderTFValue = (val: string): JSX.Element => {
...
+            {typeof objValue === "string" ? (
+              <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3">
+                <div className="text-left text-faint col-span-1">{objKey}</div>
+                <div className="col-span-2 sm:col-span-1 break-words">
+                  {renderTFValue(objValue)}
+                </div>
+              </div>
+            ) : (
+              <></>
+            )}
```
Comment (acolver) [inline radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx]: Shouldn't we just call `renderTFValues` again (recursively)? This should remove duplicated code for rendering strings as well as handle nested objects... Thoughts?

Comment (saurabhkg891) [reply radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx]: if not string then only we can call again and updated like that

File: radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx [hunk 5]
```diff
@@ -19,8 +20,8 @@ interface ModuleOutputsProps {
 
 const isURL = (val: string) => val.search(/^https?:\/\//) > -1
 
-const renderTFValue = (val: string): JSX.Element => {
-  if (isURL(val)) {
+const renderTFValue = (val: any): JSX.Element => {
```
Comment (acolver) [inline radlab-ui/webapp/src/components/outputs/ModuleOutputs.tsx:23]: Shouldn't this be `unknown` and not `any`?

**Review by acolver — APPROVED** [review]

**Pull request merged**